### PR TITLE
Fix circt-doc target

### PR DESCRIPTION
**Pull request by xgupta**
Running `make circt-doc` giving copying errors. I am not sure the exact solution but I think GEN_DOC_FILE variable is not correctly set. So I am copying the logic from MLIR's AddMLIR.cmake .

### CODE REVIEW

File: cmake/modules/AddCIRCT.cmake [hunk 1]
```diff
@@ -11,7 +11,16 @@ function(add_circt_interface interface)
 endfunction()
 
 function(add_circt_doc doc_filename command output_file output_directory)
-  add_mlir_doc(${ARGV})
+  set(LLVM_TARGET_DEFINITIONS ${doc_filename}.td)
+  tablegen(MLIR ${output_file}.md ${command} "-I${MLIR_MAIN_INCLUDE_DIR}" "-I${MLIR_INCLUDE_DIR}")
```
Comment (stephenneuendorffer) [inline cmake/modules/AddCIRCT.cmake:15]: Seems like we'll need something like CIRCT_INCLUDE_DIR too?

Comment (xgupta) [reply cmake/modules/AddCIRCT.cmake]: Added, but it was working without that also.

**Comment by xgupta** [comment]
@youngar Can you take a look at this PR? I see AddCIRCT.cmake was added by you. Is this the correct way to fix circt-doc target?

**Comment by youngar** [comment]
@xgupta LGTM and I tested it out locally.  Thanks for doing this!

I'm not sure why there are include paths on that one line.  For what its worth, I tried not using any include paths and it still worked fine.  I also notice that none of the other commands in AddMLIR.cmake are using include paths. 🤷 
```diff
+ mlir_tablegen( ${output_file}.md ${command})
- tablegen(MLIR ${output_file}.md ${command} "-I${MLIR_MAIN_INCLUDE_DIR}" "-I${MLIR_INCLUDE_DIR}")
```

Since I don't know why it is the way it is, I think your change is good to go as is 👍

**Comment by youngar** [comment]
I did a bit of digging, it looks like how tablegen include paths work was changed in `0c0831f74b7181268a777a39ee087c9337ffa0c5`. I think include paths are no longer needed as long as there is a directory-wide `include_directories(...)` invocation.  I'm thinking that the include paths for  `add_mlir_doc` and `add_circt_doc` commands are no longer needed.  @stephenneuendorffer Since you were the one to make the change, do you have any idea if this was just an oversight?

**Comment by xgupta** [comment]
> I'm not sure why there are include paths on that one line. For what its worth, I tried not using any include paths and it still worked fine. I also notice that none of the other commands in AddMLIR.cmake are using include paths. 

Thanks for reviewing. I think removing includes from AddMLIR.cmake's mlir_doc missed from the [patch](https://reviews.llvm.org/D77156). I open a Pharicator review to remove from there also.

Merging now.

**Pull request merged**